### PR TITLE
add typed/file/glob

### PR DESCRIPTION
**Pull request by bennn**
Typed interface to `file/glob`.

This PR is "ready to go", but if we leave it open I'll try to add types + tests for the other `file/*` libraries by the end of the month.

(Otherwise I'll still try to add those, just in a different PR)

### CODE REVIEW

File: typed-racket-more/typed/file/glob.rkt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+(define-type Glob (U Path-String (Sequenceof Path-String)))
+
+(unsafe-require/typed file/glob
+  [glob (->* [Glob] [#:capture-dotfiles? Boolean] (Listof Path-String))]
+  [in-glob (->* [Glob] [#:capture-dotfiles? Boolean] (Sequenceof Path-String))]
+  [glob-match? (->* [Glob Path-String] [#:capture-dotfiles? Boolean] Boolean)]
+  [glob-capture-dotfiles? (Parameterof Boolean)])
+
+(unsafe-provide glob
```
Comment (samth) [inline typed-racket-more/typed/file/glob.rkt]: We definitely don't need both `unsafe-require` and `unsafe-provide`. Presumably no one will use this library from untyped code.

File: typed-racket-more/typed/file/glob.rkt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#lang typed/racket/base
+
+(require typed/racket/unsafe)
+
+(define-type Glob (U Path-String (Sequenceof Path-String)))
+
+(unsafe-require/typed file/glob
```
Comment (samth) [inline typed-racket-more/typed/file/glob.rkt]: Does this need to be unsafe?

Comment (bennn) [inline typed-racket-more/typed/file/glob.rkt]: No, but the functions already have contracts.

**Comment by samth** [comment]
Looks good to merge.

**Pull request merged**